airflow/providers/amazon/aws/example_dags/example_dms.py

```diff
@@ -0,0 +1,347 @@
+from airflow.operators.python import get_current_context
+from airflow.providers.amazon.aws.operators.dms import (
+    DmsCreateTaskOperator,
+    DmsDeleteTaskOperator,
+    DmsDescribeTasksOperator,
+    DmsStartTaskOperator,
+    DmsStopTaskOperator,
+)
+from airflow.providers.amazon.aws.sensors.dms import DmsTaskBaseSensor, DmsTaskCompletedSensor
+
+S3_BUCKET = os.getenv('S3_BUCKET', 's3_bucket_name')
+ROLE_ARN = os.getenv('ROLE_ARN', 'arn:aws:iam::1234567890:role/s3_target_endpoint_role')
+
+# The project name will be used as a prefix for various entity names.
+# Use either PascalCase or camelCase.  While some names require kebab-case
+# and others require snake_case, they all accept mixedCase strings.
+PROJECT_NAME = 'DmsDemo'
+
+# Config values for setting up the "Source" database.
+RDS_ENGINE = 'postgres'
+RDS_PROTOCOL = 'postgresql'
+RDS_USERNAME = 'username'
+# NEVER store your production password in plaintext in a DAG like this.
+# Use Airflow Secrets or a secret manager for this in production.
+RDS_PASSWORD = 'rds_password'
+
+# Config values for RDS.
+RDS_INSTANCE_NAME = f'{PROJECT_NAME}-instance'
+RDS_DB_NAME = f'{PROJECT_NAME}_source_database'
+
+# Config values for DMS.
+DMS_REPLICATION_INSTANCE_NAME = f'{PROJECT_NAME}-replication-instance'
+DMS_REPLICATION_TASK_ID = f'{PROJECT_NAME}-replication-task'
+SOURCE_ENDPOINT_IDENTIFIER = f'{PROJECT_NAME}-source-endpoint'
+TARGET_ENDPOINT_IDENTIFIER = f'{PROJECT_NAME}-target-endpoint'
+
+# Sample data.
+TABLE_NAME = f'{PROJECT_NAME}-table'
+TABLE_HEADERS = ['apache_project', 'release_year']
+SAMPLE_DATA = [
+    ('Airflow', '2015'),
+    ('OpenOffice', '2012'),
+    ('Subversion', '2000'),
+    ('NiFi', '2006'),
+]
+TABLE_DEFINITION = {
+    'TableCount': '1',
+    'Tables': [
+        {
+            'TableName': TABLE_NAME,
+            'TableColumns': [
+                {
+                    'ColumnName': TABLE_HEADERS[0],
+                    'ColumnType': 'STRING',
+                    'ColumnNullable': 'false',
+                    'ColumnIsPk': 'true',
+                },
+                {"ColumnName": TABLE_HEADERS[1], "ColumnType": 'STRING', "ColumnLength": "4"},
+            ],
+            'TableColumnsTotal': '2',
+        }
+    ],
+}
+TABLE_MAPPINGS = {
+    'rules': [
+        {
+            'rule-type': 'selection',
+            'rule-id': '1',
+            'rule-name': '1',
+            'object-locator': {
+                'schema-name': 'public',
+                'table-name': TABLE_NAME,
+            },
+            'rule-action': 'include',
+        }
+    ]
+}
+
+
+def _create_rds_instance():
+    print('Creating RDS Instance.')
+
+    rds_client = boto3.client('rds')
+    rds_client.create_db_instance(
+        DBName=RDS_DB_NAME,
+        DBInstanceIdentifier=RDS_INSTANCE_NAME,
+        AllocatedStorage=20,
+        DBInstanceClass='db.t3.micro',
+        Engine=RDS_ENGINE,
+        MasterUsername=RDS_USERNAME,
+        MasterUserPassword=RDS_PASSWORD,
```

Comment:
Don't we have operator in Rds for this?

Reply:
Nope, not that I can tell.  There is Create/Copy/Delete Snapshot,  Start/Cancel Export, and Create/Delete Event Subscription.

Reply:
Example dags should be simple this done feels a little overwhelming it has many code lines that in fact are not really part of the essence of the example but more of the setups needed to make the point.
It feels like we should have proper operators for all the "setup" part. Consider adding the needed operators in a separate PR and then adjust this PR with updated code.
This is just an observation - if you feel comfortable with it as is... I'm happy to proceed.

Reply:
We can add those new operators, but that shouldn't block documenting the existing stuff.   The support code is there because we are transitioning it all over to System Tests which need to be self-sufficient.  These can't just be the most basic versions anymore if they are going to be used as end to end tests, they'll need to create and destroy the required resources to actually work.

I can add the new operators to my backlog list, and make updating this part of that task, but I would rather get this merged in and update it later than shelf this until that gets done.